### PR TITLE
[Ecto] Bump Ecto to ~> 3.2

### PR DESCRIPTION
**Pull request by mskv**
Behaviour `Ecto.Type` now requires more callbacks and supplies defaults with `__using__` macro.

### CODE REVIEW

File: mix.exs [hunk 1]
```diff
@@ -30,35 +30,43 @@ defmodule Bow.Mixfile do
...
-      {:tesla,    "~> 0.7", github: "teamon/tesla", branch: "master"},
-
-      {:ecto,       ">= 2.0.0 and < 2.2.0", optional: true},
-      {:erlexec,    "~> 1.7.0", optional: true},
-      {:ex_aws,     "~> 2.0", optional: true},
-      {:ex_aws_s3,  "~> 2.0", optional: true},
-      {:sweet_xml,  "~> 0.6.5", optional: true},
+      {:plug, "~> 1.0"},
+      {:tesla, "~> 0.7", github: "teamon/tesla", branch: "master"},
+      {:ecto, "~> 3.2.0", optional: true},
```
Comment (hodak) [inline mix.exs]: Isn't `~> 3.2.0` too restrictive for a library? Maybe just `>= 3.2.0`?

Comment (mentero) [reply mix.exs]: If you are doing `>=` you should also add `< 4`. With `~>` you can do `3.2` and it should accept patch releases

Comment (hodak) [reply mix.exs]: `~> 3.2` sounds responsible

Comment (mentero) [reply mix.exs]: https://hexdocs.pm/elixir/Version.html

Ok. It works a bit differently than I thought. `~> 3.2` == `>= 3.2.0 and < 4.0.0` where `~> 3.2.0` == `>= 3.2.0 and < 3.3.0`

I am ok with both of them.

Comment (mskv) [reply mix.exs]: I'd also prefer to choose `~> 3.2`, but the Ecto change that broke the old solution (i.e. extension of Ecto.Type behaviour definition) happened between `3.1.0` and `3.2.0`. That's why I went for `~> 3.2.0` here. No strong feelings about it though 🤔 . To be fair, it's just an `optional` dependency that's used purely for testing, so it does not affect library users.

Comment (mskv) [reply mix.exs]: Changed to `~> 3.2`

**Pull request merged**